lightning/backend/local.go

```diff
@@ -0,0 +1,1159 @@
+	localMinPDVersion  = *semver.New("4.0.0")
+)
+
+// Range record start and end key for localStoreDir.DB
+// so we can write it to tikv in streaming
+type Range struct {
+	start  []byte
+	end    []byte
+	length int
+}
+
+type localFileMeta struct {
+	Ts        uint64 `json:"ts"`
+	Length    int64  `json:"length"`
+	TotalSize int64  `json:"total_size"`
+}
+
+type LocalFile struct {
+	localFileMeta
+	db   *pebble.DB
+	Uuid uuid.UUID
+}
+
+func (e *LocalFile) Close() error {
+	return e.db.Close()
+}
+
+// Cleanup remove meta and db files
+func (e *LocalFile) Cleanup(dataDir string) error {
+	metaPath := filepath.Join(dataDir, e.Uuid.String()+engineMetaFileSuffix)
+	if _, err := os.Stat(metaPath); err == nil {
+		err = os.Remove(metaPath)
+		if err != nil {
+			return err
+		}
+	}
+
+	dbPath := filepath.Join(dataDir, e.Uuid.String())
+	return os.RemoveAll(dbPath)
+}
+
+type grpcClis struct {
+	mu   sync.Mutex
+	clis map[uint64]*grpc.ClientConn
+}
+
+type local struct {
+	engines  sync.Map
+	grpcClis grpcClis
+	splitCli split.SplitClient
+	tls  *common.TLS
+	pdAddr 	string
+
+	localStoreDir   string
+	regionSplitSize int64
+
+	rangeConcurrency  *worker.Pool
+	batchWriteKVPairs int
+	checkpointEnabled bool
+}
+
+// NewLocal creates new connections to tikv.
+func NewLocalBackend(
+	ctx context.Context,
+	tls *common.TLS,
+	pdAddr string,
+	regionSplitSize int64,
+	localFile string,
+	rangeConcurrency int,
+	sendKVPairs int,
+	enableCheckpoint bool,
+) (Backend, error) {
+	pdCli, err := pd.NewClient([]string{pdAddr}, tls.ToPDSecurityOption())
+	if err != nil {
+		return MakeBackend(nil), errors.Annotate(err, "construct pd client failed")
+	}
+	splitCli := split.NewSplitClient(pdCli, tls.TransToTlsConfig())
+
+	shouldCreate := true
+	if enableCheckpoint {
+		if info, err := os.Stat(localFile); err != nil {
+			if !os.IsNotExist(err) {
+				return MakeBackend(nil), err
+			}
+		} else if info.IsDir() {
+			shouldCreate = false
+		}
+	}
+
+	if shouldCreate {
+		err = os.Mkdir(localFile, 0700)
+		if err != nil {
+			return MakeBackend(nil), err
+		}
+	}
+
+	local := &local{
+		engines:  sync.Map{},
+		splitCli: splitCli,
+		tls:  	  tls,
+		pdAddr:pdAddr,
+
+		localStoreDir:   localFile,
+		regionSplitSize: regionSplitSize,
+
+		rangeConcurrency:  worker.NewPool(ctx, rangeConcurrency, "range"),
+		batchWriteKVPairs: sendKVPairs,
+		checkpointEnabled: enableCheckpoint,
+	}
+	local.grpcClis.clis = make(map[uint64]*grpc.ClientConn)
+	return MakeBackend(local), nil
+}
+
+func (local *local) getGrpcConnLocked(ctx context.Context, storeID uint64) (*grpc.ClientConn, error) {
+
+	store, err := local.splitCli.GetStore(ctx, storeID)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	opt := grpc.WithInsecure()
+	if local.tls.TransToTlsConfig() != nil {
+		opt = grpc.WithTransportCredentials(credentials.NewTLS(local.tls.TransToTlsConfig()))
+	}
+	ctx, cancel := context.WithTimeout(ctx, dialTimeout)
+	keepAlive := 10
+	keepAliveTimeout := 3
```

Comment:
```suggestion
	keepAlive := 10 * time.Second
	keepAliveTimeout := 3 * time.Second
```

we better, move these to the const section.